### PR TITLE
Fixes a cursor offset mistake for Material on iOS

### LINKED ISSUE #27626: Material TextField's cursor seems off
I think we got the CupertinoTextField right. Though the offset seems off for the Material TextField

flutter_05

flutter_02

cc @jslavitz

### PR DESCRIPTION
**Pull request by jslavitz**
Fixes #27626.

### CODE REVIEW

File: packages/flutter/test/material/text_field_test.dart [hunk 1]
```diff
@@ -3318,6 +3319,75 @@ void main() {
     semantics.dispose();
   });
 
+  testWidgets('Floating cursor is painted', (WidgetTester tester) async {
```
Comment (xster) [inline packages/flutter/test/material/text_field_test.dart]: Floating cursor is painted on iOS

File: packages/flutter/test/material/text_field_test.dart [hunk 2]
```diff
@@ -3318,6 +3319,75 @@ void main() {
...
+    debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
```
Comment (xster) [inline packages/flutter/test/material/text_field_test.dart]: Use ThemeData.platform instead of changing the platform at the root level.

**Review by xster — COMMENTED** [review]
Ok to do in a separate PR but we should have golden tests for the cursor on Material TextField on Android and iOS and for CupertinoTextField

File: packages/flutter/test/material/text_field_test.dart [hunk 3]
```diff
@@ -14,6 +14,7 @@ import 'package:flutter/services.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/gestures.dart' show DragStartBehavior;
 
+import '../rendering/mock_canvas.dart';
```
Comment (xster) [inline packages/flutter/test/material/text_field_test.dart]: This is a bit unexpected. You imported a file but didn't change anything else?

Comment (xster) [reply packages/flutter/test/material/text_field_test.dart]: actually this probably should be here, but add a test in this file that checks for the drawRRect called for Material TextField on iOS

**Comment by xster** [comment]
LGTM modulo one more test

File: packages/flutter/test/widgets/editable_text_cursor_test.dart [hunk 4]
```diff
@@ -538,55 +540,41 @@ void main() {
...
+    editableTextState.updateFloatingCursor(RawFloatingCursorPoint(state: FloatingCursorDragState.Update, offset: const Offset(20, 20)));
     await tester.pump();
 
     expect(editable, paints
-      ..rrect(
-        rrect: RRect.fromRectAndRadius(
-          Rect.fromLTRB(464.6666564941406, 2.0833332538604736, 466.6666564941406, 14.083333015441895),
-          const Radius.circular(2.0),
-        ),
+      ..rrect(rrect: RRect.fromRectAndRadius(
```
Comment (HansMuller) [inline packages/flutter/test/widgets/editable_text_cursor_test.dart]: Quite a lot of needless code reformatting here. In this case (and others) the new version matches what's explicitly called out as "bad" in the style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#if-you-have-a-newline-after-some-opening-punctuation-match-it-on-the-closing-punctuation

**Comment by xster** [comment]
LGTM. We can do this and add the test later to unblock the roll.

File: packages/flutter/test/material/text_field_test.dart [hunk 5]
```diff
@@ -4326,14 +4326,14 @@ void main() {
 
       expect(lastCharEndpoint.length, 1);
       // The last character is now on screen.
-      expect(lastCharEndpoint[0].point.dx, moreOrLessEquals(797.3333129882812));
+      expect(lastCharEndpoint[0].point.dx, moreOrLessEquals(798.6666870117188));
```
Comment (HansMuller) [inline packages/flutter/test/material/text_field_test.dart]: Tests with values like this don't really inspire confidence. This PR is only an update so I suppose there's no reason to fix the problem here. In a future PR we should either provide a rationale for these values (why is this value expected) or replace them with golden image tests.

**Review by HansMuller — APPROVED** [review]
LGTM (please undo the reformatting)

**Comment by jslavitz** [comment]
@HansMuller What reformat do you mean?

**Comment by tvolkert** [comment]
> Ok to do in a separate PR but we should have golden tests for the cursor on Material TextField on Android and iOS and for CupertinoTextField

I want to make sure this isn't dropped - we need pre-commit golden image tests that would have caught this.

**Pull request merged**